### ISSUE TITLE
Training FIFO Model and Evaluation Inquiry 

### ISSUE THREAD

**Issue by gyuwonchoi** [sohyun-l/fifo#4]
Hi,

Thank you for sharing the code of your work.

I followed the training process on NVIDIA RTX A6000 GPU as below:

`$ python main.py --file-name 'FIFO_model'   --restore-from '.pretrained/Cityscapes_pretrained_model.pth' --restore-from-fogpass './pretrained/FogPassFilter_pretrained.pth' --modeltrain 'train' 
`

And I got result of evaluation on Foggy Zurich, Foggy Driving Dense, and Cityscapes Lindau:

 pth name: FIFO2000.pth,	 Foggy Zurich: 39.56	 Foggy Driving Dense: 38.16	 Foggy Driving: 47.57	 Cityscapes lindau: 67.06
 pth name: FIFO4000.pth,	 Foggy Zurich: 41.91	 Foggy Driving Dense: 35.25	 Foggy Driving: 47.07	 Cityscapes lindau: 68.89
 pth name: FIFO6000.pth,	 Foggy Zurich: 42.66	 Foggy Driving Dense: 36.4	 Foggy Driving: 46.3	 Cityscapes lindau: 68.45
 pth name: FIFO8000.pth,	 Foggy Zurich: 44.93	 Foggy Driving Dense: 37.8	 Foggy Driving: 47.38	 Cityscapes lindau: 60.95
 pth name: FIFO10000.pth,	 Foggy Zurich: 42.11	 Foggy Driving Dense: 36.93	 Foggy Driving: 47.48	 Cityscapes lindau: 67.99
 pth name: FIFO12000.pth,	 Foggy Zurich: 44.97	 Foggy Driving Dense: 38.88	 Foggy Driving: 48.65	 Cityscapes lindau: 63.42
 pth name: FIFO14000.pth,	 Foggy Zurich: 43.95	 Foggy Driving Dense: 36.22	 Foggy Driving: 46.08	 Cityscapes lindau: 68.48
 pth name: FIFO16000.pth,	 Foggy Zurich: 43.45	 Foggy Driving Dense: 36.41	 Foggy Driving: 46.17	 Cityscapes lindau: 72.79
 pth name: FIFO18000.pth,	 Foggy Zurich: 45.52	 Foggy Driving Dense: 39.41	 Foggy Driving: 47.76	 Cityscapes lindau: 67.81
 pth name: FIFO20000.pth,	 Foggy Zurich: 43.78	 Foggy Driving Dense: 35.51	 Foggy Driving: 46.02	 Cityscapes lindau: 67.63
 pth name: FIFO25000.pth,	 Foggy Zurich: 40.43	 Foggy Driving Dense: 39.66	 Foggy Driving: 47.84	 Cityscapes lindau: 61.55
 pth name: FIFO30000.pth,	 Foggy Zurich: 40.05	 Foggy Driving Dense: 39.42	 Foggy Driving: 46.0	 Cityscapes lindau: 67.57
 pth name: FIFO35000.pth,	 Foggy Zurich: 42.91	 Foggy Driving Dense: 40.02	 Foggy Driving: 46.79	 Cityscapes lindau: 66.17
 pth name: FIFO40000.pth,	 Foggy Zurich: 46.28	 Foggy Driving Dense: 39.63	 Foggy Driving: 47.21	 Cityscapes lindau: 66.35
 pth name: FIFO45000.pth,	 Foggy Zurich: 46.30	 Foggy Driving Dense: 38.57	 Foggy Driving: 45.88	 Cityscapes lindau: 68.62
 pth name: FIFO50000.pth,	 Foggy Zurich: 42.95	 Foggy Driving Dense: 39.42	 Foggy Driving: 46.0	 Cityscapes lindau: 68.14
 pth name: FIFO55000.pth,	 Foggy Zurich: 47.23	 Foggy Driving Dense: 36.35	 Foggy Driving: 45.2	 Cityscapes lindau: 66.59
 pth name: FIFO60000.pth,	 Foggy Zurich: 44.85	 Foggy Driving Dense: 38.54	 Foggy Driving: 45.64	 Cityscapes lindau: 68.90
 pth name: FIFO65000.pth,	 Foggy Zurich: 47.59	 Foggy Driving Dense: 40.71	 Foggy Driving: 46.79	 Cityscapes lindau: 66.45
 pth name: FIFO70000.pth,	 Foggy Zurich: 45.73	 Foggy Driving Dense: 38.58	 Foggy Driving: 46.89	 Cityscapes lindau: 61.37
 pth name: FIFO75000.pth,	 Foggy Zurich: 41.55	 Foggy Driving Dense: 39.5	 Foggy Driving: 45.98	 Cityscapes lindau: 69.37
 pth name: FIFO80000.pth,	 Foggy Zurich: 43.80	 Foggy Driving Dense: 40.92	 Foggy Driving: 47.55	 Cityscapes lindau: 68.76
 pth name: FIFO85000.pth,	 Foggy Zurich: 41.53	 Foggy Driving Dense: 37.28	 Foggy Driving: 45.46	 Cityscapes lindau: 60.95
 pth name: FIFO90000.pth,	 Foggy Zurich: 41.64	 Foggy Driving Dense: 37.11	 Foggy Driving: 44.7	 Cityscapes lindau: 65.16
 pth name: FIFO95000.pth,	 Foggy Zurich: 43.69	 Foggy Driving Dense: 43.97	 Foggy Driving: 48.41	 Cityscapes lindau: 64.76
 

However, the results are not good as reported in paper especially on Foggy Driving Dense dataset.
I am wondering **(1) if you have any advice for training** and **(2) the way you selected the 'FIFO_final_model'** ?

Thank you in advance.

**Comment by sohyun-l** [comment]
Hi, Thanks for your interest in our work!

There may be a variance in performance, similar to existing domain generalization models.

(1) We trained our model by fixing the initial random seed, hardware, and environment settings. We recommend using the pre-trained weight of the fog-pass filtering module we trained.

(2) We selected the best-performing model by evaluating each real foggy dataset, like existing works.
